apps/system/sekey_test/sekey_test.c

```diff
@@ -0,0 +1,239 @@
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * apps/system/sekey_test/sekey_test.c
+ *
+ *   Copyright (C) 2008, 2011-2012 Gregory Nutt. All rights reserved.
+ *   Author: Gregory Nutt <gnutt@nuttx.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <tinyara/security_hal.h>
+#include <tinyara/seclink.h>
+#include <mbedtls/x509_crt.h>
+
+#ifdef CONFIG_ECDSA_KEY_SLOT
+#define ECDSA_KEY_SLOT CONFIG_ECDSA_KEY_SLOT
+#else
+#define ECDSA_KEY_SLOT "0,"
+#endif
+
+#ifdef CONFIG_CERT_SLOT
+#define CERT_SLOT CONFIG_CERT_SLOT
+#else
+#define CERT_SLOT "1,"
+#endif
+
+static uint8_t key_test[128] = {
+	0,
+};
+static uint8_t cert_test[128] = {
+	0,
+};
+
+int sekey_test_main(int argc, char *argv[])
+{
+	int i;
+	int ret;
+	int res = 0;
+	char *ptr;
+	sl_ctx shnd;
+	hal_result_e hres = 0;
+	hal_ecdsa_mode ecdsa_mode;
+	hal_data t_hash = {
+		0,
+	};
+	int hash_len = 32;
```

Comment:
If `hash_len` is not changed in the code, how about using `#define HASH_LEN 32`?

Reply:
Done